ciscosma.json

```
@@ -0,0 +1,2297 @@
+                    "data_path": "action_result.summary.total_entries",
+                    "data_type": "numeric"
+                },
+                {
+                    "data_path": "action_result.summary.entries_returned",
+                    "data_type": "numeric"
+                },
+                {
+                    "data_path": "action_result.summary.list_type",
+                    "data_type": "string"
+                },
+                {
+                    "data_path": "summary.total_objects",
+                    "data_type": "numeric"
+                },
+                {
+                    "data_path": "summary.total_objects_successful",
+                    "data_type": "numeric"
+                },
+                {
+                    "data_path": "action_result.message",
+                    "data_type": "string"
+                }
+            ],
+            "render": {
+                "type": "table"
+            },
+            "versions": "EQ(*)"
+        },
+        {
+            "action": "add list entry",
+            "identifier": "add_list_entry",
+            "description": "Add an entry to the spam quarantine safelist or blocklist",
+            "type": "generic",
+            "read_only": false,
+            "parameters": {
+                "list_type": {
+                    "description": "Type of list to add entry to",
+                    "data_type": "string",
+                    "required": true,
+                    "order": 0,
+                    "value_list": ["safelist", "blocklist"],
+                    "default": "safelist"
+                },
+                "view_by": {
+                    "description": "Add entry by sender or recipient",
+                    "data_type": "string",
+                    "order": 1,
+                    "value_list": ["sender", "recipient"],
+                    "default": "recipient"
+                },
+                "recipient_addresses": {
+                    "description": "Recipient email addresses (comma-separated if multiple) - Required with view_by=recipient",
+                    "data_type": "string",
+                    "order": 2
+                },
+                "sender_list": {
+                    "description": "Sender addresses or domains (comma-separated if multiple) - Required with view_by=recipient",
+                    "data_type": "string",
+                    "order": 3
+                },
+                "sender_addresses": {
+                    "description": "Sender addresses or domains (comma-separated if multiple) - Required with view_by=sender",
+                    "data_type": "string",
+                    "order": 4
+                },
+                "recipient_list": {
+                    "description": "Recipient email addresses (comma-separated if multiple) - Required with view_by=sender",
+                    "data_type": "string",
+                    "order": 5
+                }
```

I'm not sure I understand these parameter choices.

- What is the difference between `*_addresses` and `*_list`?
- Could this not have been flattened/simplified with only two required parameters?
```suggestion
                "recipients": {
                    "description": "Recipient email addresses or domains (comma-separated if multiple)",
                    "required": true,
                    "data_type": "string",
                    "order": 2
                },
                "senders": {
                    "description": "Sender addresses or domains (comma-separated if multiple)",
                    "required": true,
                    "data_type": "string",
                    "order": 3
                }
```

---

This is unnecessary complexity, but it also provides explicitness with the Cisco SMA API to keep things consistent on both sides, as they use these fields. We could simplify it and handle the split internally, but that would remove some clarity. [Cisco SMA API Fields](https://www.cisco.com/c/en/us/td/docs/security/security_management/sma/sma15-5-1/api_guide/b_sma_api_guide_15_5_1/test_chapter_010.html#:~:text=edited.%20The%20syntax%20and%20supported%20attributes%20are%20given%20below%3A-,Synopsis,Content%2DType%2C%20Content%2DLength%2C%20Connection,-Examples)

---

I suppose it's fine and consistent with our existing philosophy of app development, so I'll let it through. But I think we might want to reconsider that philosophy long-term